src/plugins/passwd/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+include (LibAddMacros)
+try_compile(HAS_FGETPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasfgetpwent.c")
+if(HAS_FGETPWENT)
+    add_definitions(-DHAS_FGETPWENT)
+endif()
+try_compile(HAS_PUTPWENT "${CMAKE_BINARY_DIR}" "${PROJECT_SOURCE_DIR}/src/plugins/passwd/hasputpwent.c")
+if(HAS_PUTPWENT)
+    add_definitions(-DHAS_PUTPWENT)
```

Comment:
You do not need an `if` statement, its should be ok if no string (empty string) is after  `COMPILE_DEFINITIONS`.